### DISCUSSION TITLE
Can't use extensions from another module

### DISCUSSION THREAD

**Discussion by Sinan-Karakaya** [skiptools/skip#277] [Q&A] [answered]
I created a multi-module app following the documentation, with one of the module being the app, and the other a design system containing extensions of components like so:

```swift
public struct PrimaryButtonStyle: ButtonStyle {
    public init() {}
    
    public func makeBody(configuration: Configuration) -> some View {
        configuration.label
          .fontWeight(Font.Weight.bold)
          .font(Font.title3)
          .fontDesign(Font.Design.rounded)
          .padding()
          .foregroundStyle(Color.white)
          .background(configuration.isPressed ? Color.gray.opacity(0.2) : Color.accentColor)
          .clipShape(RoundedRectangle(cornerRadius: 12))
    }
}

public extension ButtonStyle where Self == PrimaryButtonStyle {
    static var primary: PrimaryButtonStyle {
        PrimaryButtonStyle()
    }
}
```

But when I try to use it like so in the Skip app, I get a bunch of the following errors:

```swift
import MyModule

Button("Hey there!") {
    // noop
}
.buttonStyle(.primary)
```

```
DesignSystem.kt:3:17 Unresolved reference 'compose'.
DesignSystem.kt:4:17 Unresolved reference 'compose'.
DesignSystem.kt:5:17 Unresolved reference 'compose'.
DesignSystem.kt:6:17 Unresolved reference 'compose'.
DesignSystem.kt:7:17 Unresolved reference 'compose'.
DesignSystem.kt:8:17 Unresolved reference 'compose'.
DesignSystem.kt:9:17 Unresolved reference 'compose'.
DesignSystem.kt:12:13 Unresolved reference 'ui'.
DesignSystem.kt:14:13 Unresolved reference 'model'.
DesignSystem.kt:24:33 Unresolved reference 'ButtonStyle'.
DesignSystem.kt:28:33 Unresolved reference 'Configuration'.
DesignSystem.kt:28:49 Unresolved reference 'View'.
DesignSystem.kt:29:30 Unresolved reference 'label'.
DesignSystem.kt:30:25 Unresolved reference 'Font'.
DesignSystem.kt:31:19 Unresolved reference 'Font'.
DesignSystem.kt:32:25 Unresolved reference 'Font'.
DesignSystem.kt:34:30 Unresolved reference 'Color'.
DesignSystem.kt:35:43 Unresolved reference 'isPressed'.
DesignSystem.kt:35:54 Unresolved reference 'Color'.
DesignSystem.kt:35:83 Unresolved reference 'Color'.
DesignSystem.kt:36:24 Unresolved reference 'RoundedRectangle'.
```

And inside the design system module itself I get those 2 errors:

```
DesignSystem.swift:1:1 Unresolved reference 'ui'.
DesignSystem.swift:6:1 Unresolved reference 'ButtonStyle'.
```

I am not sure where this issue comes from, as I didn't really find any similar issues online. Any help is appreciated.

**Comment by marcprux** [comment]
I think you just need to `import SwiftUI` at the top of the file.

**Reply by aabewhite** [reply, marked as answer]
Sorry, Skip doesn't support button styles. You can view the list of supported SwiftUI here: https://skip.tools/docs/modules/skip-ui/#supported-swiftui

**Reply by Sinan-Karakaya** [reply]
That's too bad :/ Is it a quite easy feature to implement in Skip ? I'd be fine trying to implement it

**Comment by aabewhite** [comment]
Unfortunately it would require modifications to our transpiler as well in order to properly recognize the makeBody function as a @ViewBuilder. We haven't had many requests for ButtonStyle support - I assume most users are just creating their own View type whose body displays a Button whose label is styled, and then using that instead

**Reply by Sinan-Karakaya** [reply]
I guess I could do that :/ I just wished it could have been done that way but It's fine !

**Reply by aabewhite** [reply]
It is certainly something we'd like to support in the future, we just can't commit to a specific timeframe right now
